### PR TITLE
`LinearOperator.solve` method

### PR DESCRIPTION
**Pull request by marvinpfoertner**
# In a Nutshell
This PR adds the method `solve` to the `LinearOperator` interface.

# Detailed Description
- `LinearOperator.solve`
- `_InverseLinearOperator` uses `LinearOperator.solve` as `_matmul`
- remove redundant `rmatmul` argument in `LambdaLinearOperator`

# Related Issues
None

### CODE REVIEW

File: src/probnum/linops/_arithmetic_fallbacks.py [hunk 1]
```diff
@@ -227,6 +223,9 @@ def __repr__(self):
             res += f"\t{s}, \n"
         return res + "]"
 
+    def _solve(self, B: np.ndarray) -> np.ndarray:
+        return functools.reduce(lambda vec, op: op @ vec, self._factors, B)
```
Comment (JonathanWenger) [inline src/probnum/linops/_arithmetic_fallbacks.py:227]: I may be misunderstanding, but doesn't this line compute $(B F_1) F_2$, where the linear operator is given by $L = F_1 F_2$? Shouldn't it rather be something like 
```suggestion
        return functools.reduce(lambda rhs, op: op.solve(rhs), self._factors, B)
```
If yes, also add a test for the product linear operator.

File: src/probnum/linops/_linear_operator.py [hunk 2]
```diff
@@ -172,6 +174,97 @@ def __call__(self, x: np.ndarray, axis: Optional[int] = None) -> np.ndarray:
         else:
             raise ValueError("The operand must be at least one dimensional.")
 
+    def solve(self, b: ArrayLike) -> np.ndarray:
+        """Solves linear systems :code:`A @ x = b`, where :code:`b` is either a vector
+        or a (stack of) matrices.
+
+        This method behaves exactly like :code:`A.inv() @ b`.
```
Comment (JonathanWenger) [inline src/probnum/linops/_linear_operator.py:181]: Do we want to guarantee this? This is generally not the case for numpy arrays and will also not be the case anymore once we allow the user to choose a linear solver. I suggest not promising this in the docstring.
```suggestion
```

**Review by JonathanWenger — CHANGES_REQUESTED** [review]
Mostly looks good. Only the implementation of the solve method for the `ProductLinearOperator` needs a second look I think.

File: src/probnum/linops/_arithmetic_fallbacks.py (hunk 1 above)
Comment (marvinpfoertner) [reply src/probnum/linops/_arithmetic_fallbacks.py]: Oops, that was kind of a last minute addition. I wonder why codecov did not catch the missing coverage?

File: src/probnum/linops/_linear_operator.py (hunk 2 above)
Comment (marvinpfoertner) [reply src/probnum/linops/_linear_operator.py]: I changed the docstring to communicate what I actually meant with this line in the first place.

**Review by JonathanWenger — APPROVED** [review]
👍👍

**Comment by marvinpfoertner** [comment]
GitHub Actions seems to have network issues when communicating with its apt mirror. I already restarted the `run_notebooks` action multiple times in this PR and in #776.
If it's alright with you, I will bypass branch protections and merge anyway. The same also applies to #776.

**Comment by JonathanWenger** [comment]
> GitHub Actions seems to have network issues when communicating with its apt mirror. I already restarted the `run_notebooks` action multiple times in this PR and in #776. If it's alright with you, I will bypass branch protections and merge anyway. The same also applies to #776.

Others are reporting similar failures. See e.g. https://github.com/actions/runner-images/issues/675

**Pull request merged**